rmw_cyclonedds_cpp/src/serdata.cpp

```diff
@@ -434,9 +438,9 @@ static void sertopic_rmw_free(struct ddsi_sertopic * tpcmn)
 #endif
   if (tp->type_support.type_support_) {
     if (using_introspection_c_typesupport(tp->type_support.typesupport_identifier_)) {
-      delete static_cast<MessageTypeSupport_c *>(tp->type_support.type_support_);
+      delete static_cast<TypeSupport_c *>(tp->type_support.type_support_);
     } else if (using_introspection_cpp_typesupport(tp->type_support.typesupport_identifier_)) {
-      delete static_cast<MessageTypeSupport_cpp *>(tp->type_support.type_support_);
+      delete static_cast<TypeSupport_cpp *>(tp->type_support.type_support_);
```

Comment:
If `Typesupport` would have a non templated abstract base class with a virtual destructor all this could be reduced to:

```cpp
delete static_cast<TypeSupportBase *>(tp->type_support.type_support_);
```

Reply:
this can be ignored for the moment